Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleWindow.cpp

```diff
@@ -37,6 +38,62 @@ namespace ShaderManagementConsole
         Base::OnDocumentOpened(documentId);
         m_documentInspector->SetDocumentId(documentId);
     }
+
+    void ShaderManagementConsoleWindow::UpdateMenus(QMenuBar* menuBar)
+    {
+        Base::UpdateMenus(menuBar);
+        m_actionSaveAsChild->setVisible(false);
+    }
+
+    AZStd::string ShaderManagementConsoleWindow::GetSaveDocumentParams(const AZStd::string& initialPath) const
+    {
+        // Get shader file path
+        AZStd::string shaderFullPath;
+        AZ::RPI::ShaderVariantListSourceData shaderVariantList = {};
+        ShaderManagementConsoleDocumentRequestBus::EventResult(
+            shaderVariantList,
+            GetCurrentDocumentId(),
+            &ShaderManagementConsoleDocumentRequestBus::Events::GetShaderVariantListSourceData);
+        shaderFullPath = AZ::RPI::AssetUtils::ResolvePathReference(initialPath, shaderVariantList.m_shaderFilePath);
+        
+        QMessageBox msgBox;
+        msgBox.setText("Where do you want to save the list?");
+        QPushButton* projectBtn = msgBox.addButton(QObject::tr("Save to project"), QMessageBox::YesRole);
+        QPushButton* engineBtn = msgBox.addButton(QObject::tr("Save to engine"), QMessageBox::NoRole);
+        msgBox.exec();
+
+        AZStd::string result;
+        if (msgBox.clickedButton() == projectBtn)
+        {
+            AZ::IO::FixedMaxPath projectPath = AZ::Utils::GetProjectPath();
+            AZStd::string relativePath, rootFolder;
+            bool pathFound;
+            AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+                pathFound,
+                &AzToolsFramework::AssetSystemRequestBus::Events::GenerateRelativeSourcePath,
+                shaderFullPath,
+                relativePath,
+                rootFolder);
+
+            if (pathFound)
+            {
+                AzFramework::StringFunc::Path::ReplaceExtension(relativePath, "shadervariantlist");
+                result = AZStd::string::format("%s\\ShaderVariants\\%s", projectPath.c_str(), relativePath.c_str());
+                AzFramework::StringFunc::Path::Normalize(result);
```

Comment:
Instead of using the StringFunc functions, the [AZ::IO::BasicPath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L388) API should be used instead which models the python pathlib API.

We aim to get remove the AZ::StringFunc::Path functions in the future.

```suggestion
                relativePath.ReplaceExtension("shadervariantlist");
                projectPath /= "ShaderVariants" / relativePath;
                projectPath = projectPath.LexicallyNormal();
```

Reply:
Thanks for the heads-up!